packages/validator/test/unit/services/attestationDuties.test.ts

```diff
@@ -241,4 +242,137 @@ describe("AttestationDutiesService", () => {
+    const setupDuty = (slot: number): routes.validator.AttesterDuty => ({
+      slot,
+      committeeIndex: 1,
+      committeeLength: 120,
+      committeesAtSlot: 120,
+      validatorCommitteeIndex: 1,
+      validatorIndex: index,
+      pubkey: pubkeys[0],
+    });
+
+    async function setupTestEnvironment(duty: routes.validator.AttesterDuty, dependentRoot = "0xOLD_ROOT") {
+      api.validator.getAttesterDuties.mockResolvedValue(
+        mockApiResponse({
+          data: [duty],
+          meta: {dependentRoot, executionOptimistic: false},
+        })
+      );
+      api.validator.prepareBeaconCommitteeSubnet.mockResolvedValue(mockApiResponse({}));
+
+      const clock = new ClockMock();
+      const syncingStatusTracker = new SyncingStatusTracker(loggerVc, api, clock, null);
+
+      let onNewHeadCallback: ((headEvent: HeadEventData) => Promise<void>) | undefined;
+      chainHeadTracker.runOnNewHead.mockImplementation((callback) => {
+        onNewHeadCallback = callback;
+      });
+
+      const dutiesService = new AttestationDutiesService(
+        loggerVc,
+        api,
+        clock,
+        validatorStore,
+        chainHeadTracker,
+        syncingStatusTracker,
+        null
+      );
+
+      if (!onNewHeadCallback) throw new Error("onNewHeadCallback not initialized");
+      return {clock, dutiesService, onNewHeadCallback: onNewHeadCallback};
+    }
+
+    it("Should resubscribe to beacon subnets when current epoch dependent root changes", async () => {
+      const slot = 5;
+      const epoch = computeEpochAtSlot(slot);
+      const duty = setupDuty(slot);
+
+      const {clock, dutiesService, onNewHeadCallback} = await setupTestEnvironment(duty);
+      await clock.tickEpochFns(epoch, controller.signal);
+
+      expect(dutiesService["dutiesByIndexByEpoch"].get(epoch)).toBeDefined();
+      expect(api.validator.prepareBeaconCommitteeSubnet).toHaveBeenCalledTimes(1);
+
+      const reorgedDuty = {...duty, slot: slot + 1, committeeIndex: 3};
+      api.validator.getAttesterDuties.mockResolvedValue(
+        mockApiResponse({
+          data: [reorgedDuty],
+          meta: {dependentRoot: "0xNEW_ROOT", executionOptimistic: false},
+        })
+      );
+
+      await onNewHeadCallback({
+        slot,
+        head: "0xHEAD_BLOCK",
+        previousDutyDependentRoot: "0xOLD_ROOT",
+        currentDutyDependentRoot: "0xNEW_ROOT",
+      });
+
+      expect(api.validator.prepareBeaconCommitteeSubnet).toHaveBeenCalledTimes(2);
+
+      const lastCallArgs = api.validator.prepareBeaconCommitteeSubnet.mock.calls[1][0];
+      expect(lastCallArgs.subscriptions).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            validatorIndex: index,
+            slot: reorgedDuty.slot,
+            committeeIndex: reorgedDuty.committeeIndex,
+          }),
+        ])
+      );
+    });
+
+    it("Should resubscribe when next epoch dependent root changes", async () => {
```

Comment:
I don't think this test does what the title here says, please also make sure `` is called as expected

but to test next epoch attester duties reorg you need to have duties at that epoch, which this test does not, it just sets up a duty for current epoch

see code here https://github.com/ChainSafe/lodestar/blob/5317389489e1553be2e98f79f0c0f8c74ff79f28/packages/validator/src/services/attestationDuties.ts#L332-L334